src/clj/puppetlabs/services/master/master_core.clj

```diff
@@ -130,8 +130,16 @@
   [master-ns :- schema/Keyword
    config-route]
   (cond
-    (and (map? config-route) (contains? config-route :master-routes))
-    (:master-routes config-route)
+    ;; if the route config is a map, we need to determine whether it's the
+    ;; new-style multi-server config (where there will be a `:route` key and a
+    ;; `:server`, key), or the old style where there is a single key that is
+    ;; assumed to be our hard-coded route id (`:master-routes`).
+    ;; It should be possible to delete this hack (perhaps this entire function)
+    ;; when we remove support for legacy routes.
+    (and (map? config-route) (or (contains? config-route :route)
```

Comment:
Do we also want to support multi-route, multi-server configs? In that case I believe it would look something like

```
{
  :web-router-service {
    :puppetlabs.services.master.master-service/master-service {
      :master-routes {
        :route "/puppet"
        :server "puppet-server"
      }
    }
  }
}
```
It would let us mount multiple routes through the master service, but if we don't plan to ever do that then this is fine as-is.